src/main/scala/extracells/api/IStorageCellAdvanced.java

```diff
@@ -0,0 +1,105 @@
+package extracells.api;
+
+
+import appeng.api.storage.ICellWorkbenchItem;
+import appeng.api.storage.data.IAEItemStack;
+import net.minecraft.item.ItemStack;
+
+import javax.annotation.Nullable;
+
+
+/**
+ * Any item which implements this can be treated as an IMEInventory via
+ * Util.getCell / Util.isCell It automatically handles the internals and NBT
+ * data, which is both nice, and bad for you!
+ * <p>
+ * Good cause it means you don't have to do anything, bad because you have
+ * little to no control over it.
+ * <p>
+ * The standard AE implementation only provides 1-63 Types
+ */
+public interface IStorageCellAdvanced extends ICellWorkbenchItem
+{
+
+	/**
+	 * It wont work if the return is not a multiple of 8.
+	 * The limit is ({@link Integer#MAX_VALUE} + 1) / 8.
+	 *
+	 * @param cellItem item
+	 * @return number of bytes
+	 */
+	long getBytes( ItemStack cellItem );
+
+	/**
+	 * Determines the number of bytes used for any type included on the cell.
+	 *
+	 * @param cellItem item
+	 * @return number of bytes
+	 * @deprecated use {@link IStorageCell#getBytesPerType(ItemStack)}
+	 */
+	@Deprecated
+	int BytePerType( ItemStack cellItem );
```

Comment:
a new interface with a deprecated member?

Reply:
Straight up just copied all the code from AE2 and changed Ints to be Longs which would explain this.

I could likely go over the functions and comments, they might be misleading too as I didn't touch them.

Reply:
EC2 code quality is not great as it is, no need to make it even more confusing I think.